include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1049 @@
+
+        @param lhs A string to compare
+        @param rhs A StringViewable to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+      */
+    template <class StringViewable>
+    friend
+#ifndef BOOST_URL_DOCS
+    typename std::enable_if<
+        std::is_constructible<string_view, StringViewable>::value, bool
+    >::type
+#else
+    bool
+#endif
+    operator!=(
+            decoded_view lhs,
+            StringViewable const& rhs) noexcept
+    {
+        return lhs.compare(string_view(rhs)) != 0;
+    }
+
+    /** Compares two strings
+
+        @note This function only participates in overload resolution
+        if we can construct a string_view from the types being
+        compared.
+
+        @tparam StringViewable A type from which we can construct a string_view
+
+        @param lhs A string to compare
+        @param rhs A StringViewable to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    template <class StringViewable>
+    friend
+#ifndef BOOST_URL_DOCS
+    typename std::enable_if<
+        std::is_constructible<string_view, StringViewable>::value, bool
+    >::type
+#else
+    bool
+#endif
+    operator<(
+            decoded_view lhs,
+            StringViewable const& rhs) noexcept
+    {
+        return lhs.compare(string_view(rhs)) < 0;
+    }
+
+    /** Compares two strings
+
+        @note This function only participates in overload resolution
+        if we can construct a string_view from the types being
+        compared.
+
+        @tparam StringViewable A type from which we can construct a string_view
+
+        @param lhs A string to compare
+        @param rhs A StringViewable to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    template <class StringViewable>
+    friend
+#ifndef BOOST_URL_DOCS
+    typename std::enable_if<
+        std::is_constructible<string_view, StringViewable>::value, bool
+    >::type
+#else
+    bool
+#endif
+    operator<=(
+            decoded_view lhs,
+            StringViewable const& rhs) noexcept
+    {
+        return lhs.compare(string_view(rhs)) <= 0;
+    }
+
+    /** Compares two strings
+
+        @note This function only participates in overload resolution
+        if we can construct a string_view from the types being
+        compared.
+
+        @tparam StringViewable A type from which we can construct a string_view
+
+        @param lhs A string to compare
+        @param rhs A StringViewable to compare
+
+        @return true if the corresponding comparison holds after percent decoding
+     */
+    template <class StringViewable>
+    friend
+#ifndef BOOST_URL_DOCS
+    typename std::enable_if<
+        std::is_constructible<string_view, StringViewable>::value, bool
+    >::type
+#else
+    bool
+#endif
+    operator>(
+            decoded_view lhs,
+            StringViewable const& rhs) noexcept
```

Comment:
odd indentation but I'm also a little confused. If we have these templates what do we need all the other `string_view` comparison overloads for? Like this one:
https://github.com/CPPAlliance/url/pull/229/files#diff-7ee7b43158ca027820d486b5a0c5b10621fe2bb4f59c17d14d2d5754202e2423R628